codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -334,7 +334,15 @@ update these file paths accordingly.
     }
 
     final Set<String> fileWithEntitlements = <String>{};
-    fileWithEntitlements.addAll(await fileSystem.file(entitlementFilePath).readAsLines());
+
+    try {
+      fileWithEntitlements.addAll(await fileSystem.file(entitlementFilePath).readAsLines());
+    } finally {
+      // TODO(xilaizhang) : add back metadata information after https://github.com/flutter/flutter/issues/126705
+      // is resolved.
+      await fileSystem.file(entitlementFilePath).delete();
+    }
```

Comment:
```suggestion
    fileWithEntitlements.addAll(await fileSystem.file(entitlementFilePath).readAsLines());
    // TODO(xilaizhang) : add back metadata information after https://github.com/flutter/flutter/issues/126705
    // is resolved.
    await fileSystem.file(entitlementFilePath).delete();
```

You don't need the `try` - `finally` if you're not doing any error handling.

Reply:
thanks for the review! umm initially I don't have the try finally block, but I was worried that the second await statement could get executed before the first await statement: i.e., the file could have got deleted before it is read. My understanding is that there might not be an enforced order among the await statements? umm I added the try finally block to enforce a sequential order

Reply:
Never mind, awaits are sequential inside the same function. TIL, thank you Christopher!